Web console: Page downloader, and fix JSON error resetting

vogievetsky opened the pull request:
Added ability download pages of results from the new async APIs:

<img width="720" alt="image" src="https://github.com/apache/druid/assets/177816/9035f176-f8a7-4b6b-94a8-3aaadded6d46">

Also fixed the issue where a JSON error might cause the Next button to stay disabled if there is a json error while typing JSON

adarshsanjeev commented:
Tried this out locally and it works great! Way better than manually making posts to get results.

I noticed something only while downloading the results, the format saved is as a json file, though according to the ResultFormat, ObjectLines is not valid json. (Firefox pointed this out too)
https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/http/ResultFormat.java#L96
Should the file extension be changed here?

I think part of the reason might be that the get results api is tagged as `@Produces(MediaType.APPLICATION_JSON)`. I think I missed this while changing the default format. I'm not sure if the console takes this into account while deciding the file extension.

adarshsanjeev commented:
Tried making the change to `@Produces(MediaType.TEXT_PLAIN)` locally to see if that fixes it, but the files downloaded are still `.json`.

I also noticed that the `Edit context` part of the UI has the plural as `X keies` in this PR (and not in the build of master), though I don't see any such change in the diff. Is this just me?
<img width="376" alt="Screenshot 2023-08-01 at 5 26 01 PM" src="https://github.com/apache/druid/assets/9818861/ba01bde5-d8c0-460d-a24b-c192e7b5f22e">

vogievetsky commented:
Thank you for the testing and the feedback @adarshsanjeev I have addressed both issues in my commit. The extension is hard coded in the console right now and I changed it to [jsonl](https://www.atatus.com/glossary/jsonl/) which is more appropriate. Also I fixed the pluralizer
